hbase-client/src/main/java/org/apache/hadoop/hbase/filter/FilterListWithAND.java

```diff
@@ -169,10 +186,14 @@ public ReturnCode filterCell(Cell c) throws IOException {
     }
     ReturnCode rc = ReturnCode.INCLUDE;
     this.seekHintFilters.clear();
-    for (int i = 0, n = filters.size(); i < n; i++) {
+    int i = 0;
+    int n = filters.size();
+    for (; i < n; i++) {
       Filter filter = filters.get(i);
       if (filter.filterAllRemaining()) {
-        return ReturnCode.NEXT_ROW;
+        rc = ReturnCode.NEXT_ROW;
```

Comment:
Let's add more comments here?

The implementation is already very complicated...

Reply:
Will do.